external-crates/move/crates/move-compiler/src/typing/core.rs

```diff
@@ -1562,10 +1638,42 @@ pub fn make_num_tvar(context: &mut Context, loc: Loc) -> Type {
     sp(loc, Type_::Var(tvar))
 }
 
+pub fn make_string_tvar(context: &mut Context, loc: Loc) -> Type {
+    let tvar = context.subst.new_string_var(&mut context.tvar_counter, loc);
+    sp(loc, Type_::Var(tvar))
+}
+
 pub fn make_tvar(context: &mut Context, loc: Loc) -> Type {
     sp(loc, Type_::Var(context.tvar_counter.next()))
 }
 
+pub fn is_u8_type(ty: &Type) -> bool {
+    match &ty.value {
+        Type_::Apply(_, sp!(_, TypeName_::Builtin(sp!(_, bt))), _) => bt.is_u8(),
+        _ => false,
+    }
+}
+
+pub fn is_inferred_string_type(name: &N::TypeName) -> bool {
```

Comment:
I have this `is` function that was mostly used for Sui mode stuff, can we reuse that here?